### PR TITLE
Add unit tests for execution results acquisition in the block synchronizer

### PR DESCRIPTION
**Pull request by alsrdn**
Unit tests for the execution results acquisition logic as part of https://github.com/casper-network/casper-node/issues/3437.

### CODE REVIEW

File: node/src/components/block_synchronizer/execution_results_acquisition/tests.rs [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+
+    let apply_result = apply_chunk(
+        *Block::random(&mut rng).hash(),
+        ExecutionResultsChecksum::Uncheckable,
+        test_chunks.clone().into_iter().collect(),
+        first_chunk.1.clone(),
+        None,
+    );
+
+    assert_matches!(apply_result, Err(Error::InvalidChunkCount { .. }));
```
Comment (rafal-ch) [inline node/src/components/block_synchronizer/execution_results_acquisition/tests.rs:81]: I think we're only missing tests for `BlockHashMismatch` and `ChunkCountMismatch` error variants. Please consider extending the tests coverage.

**Review by georgepisaltu — APPROVED** [review]
LGTM

File: node/src/components/block_synchronizer/execution_results_acquisition/tests.rs (hunk 1 above)
Comment (alsrdn) [reply node/src/components/block_synchronizer/execution_results_acquisition/tests.rs:81]: Added in https://github.com/casper-network/casper-node/pull/3698/commits/cf25d79e8b45387b0b5396422f2fa0d58627f3c4

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

**Pull request merged**